File: backend_python/src/schemas/gallery.py
from datetime import datetime

from pydantic import BaseModel


class GalleryAssetMetadata(BaseModel):
    model_config = {"extra": "forbid"}
    size: int | None = None
    uploadedBy: str
    projectId: str
    projectName: str

class GalleryAsset(BaseModel):
    model_config = {"extra": "forbid"}
    id: str
    type: str # 'image' | 'video' | 'quote' | 'unknown'
    url: str
    thumbnail: str | None = None
    title: str
    createdAt: datetime
    timestamp: datetime
    metadata: GalleryAssetMetadata

class GalleryResponse(BaseModel):
    model_config = {"extra": "ignore"}
    assets: list[GalleryAsset]
    hasMore: bool
    lastVisibleId: str | None = None

File: web_client/types/gallery.ts
export interface GalleryAssetMetadata {
    size?: number;
    uploadedBy: string;
    projectId: string;
    projectName: string;
}

export interface GalleryAsset {
    id: string;
    type: 'image' | 'video' | 'quote' | 'render' | 'unknown';

    url: string;
    thumbnail?: string;
    title: string;
    createdAt: string | Date;
    timestamp: string | Date;
    metadata: GalleryAssetMetadata;
}

export interface GalleryResponse {
    assets: GalleryAsset[];
    hasMore: boolean;
    lastVisibleId?: string;
}
